### PR TITLE
Add internal support for datatype update

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

File: src/expr/dtype_cons.cpp [hunk 1]
```diff
@@ -565,14 +558,24 @@ bool DTypeConstructor::resolve(
       }
       Trace("datatypes-init")
           << "  ...range after parametric substitution " << range << std::endl;
-      arg->d_selector = sm->mkDummySkolem(
-          argName,
-          nm->mkSelectorType(self, range),
-          "is a selector",
-          NodeManager::SKOLEM_EXACT_NAME | NodeManager::SKOLEM_NO_NOTIFY);
     }
+    arg->d_selector = sm->mkDummySkolem(
```
Comment (yoni206) [inline src/expr/dtype_cons.cpp:562]: Can you document this block? e.g., why o we need a skolem?

File: src/expr/node_manager.h [hunk 2]
```diff
@@ -34,6 +34,8 @@
 
 namespace cvc5 {
 
+using Record = std::vector<std::pair<std::string, TypeNode>>;
```
Comment (yoni206) [inline src/expr/node_manager.h:37]: can you document how this will be used?

**Review by yoni206 — APPROVED** [review]
LGTM. 2 minor comments inside. Also, before merging -- could you check if you could add a unit test? For example, that verifies that the type checker fails for ill-typed terms with the updated?

File: src/expr/node_manager.h (hunk 2 above)
Comment (ajreynol) [reply src/expr/node_manager.h:37]: This will be moved from record.h in the other PR, I'll remove from this PR.

**Pull request merged**